File: Other Preparation/cpc21c1p2.java
// code by savir singh
// https://dmoj.ca/problem/cpc21c1p2
// 100/100

// java version:

import java.util.*;

public class Main {
    static final long MM = 200001L;

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        long n = scanner.nextLong();
        n *= 2;
        n += 1;
        long ret = 0;
        List<Long> a = new ArrayList<>();
        List<Long> b = new ArrayList<>();

        for (long i = 0; i < n; i++) {
            long x = scanner.nextLong();
            ret ^= x;
            a.add(x);
        }

        for (long i = 0; i < n; i++) {
            long x = scanner.nextLong();
            ret ^= x;
            b.add(x);
        }

        for (long i = 0; i < n; i++) {
            b.set((int) i, b.get((int) i) ^ ret);
        }

        Collections.sort(a);
        Collections.sort(b);

        for (long i = 0; i < n; i++) {
            if (!a.get((int) i).equals(b.get((int) i))) {
                ret = -1;
                break;
            }
        }

        System.out.println(ret);
    }
}

File: Other Preparation/cpc21c1p2.cpp
// code by savir singh
// https://dmoj.ca/problem/cpc21c1p2
// 100/100

#include <bits/stdc++.h>
using namespace std;
#define int long long

const int MM = 2e5 + 1;
int n, ret;
vector<int> a, b;

int32_t main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cin >> n;n*=2;n+=1;
    for (int i = 0, x; i < n; i++) {
        cin >> x; ret ^= x; a.push_back(x);
    }
    for (int i = 0, x; i < n; i++) {
        cin >> x; ret ^= x; b.push_back(x);
    }
    for (int i = 0 ; i < n; i++) b[i] ^= ret;
    sort(a.begin(), a.end());sort(b.begin(), b.end());
    for (int i = 0 ; i < n; i++) {
        if (a[i] != b[i]) {ret = -1;break;}
    }
    cout << ret << endl;
}
